pkg/vault/contracts/VaultAdmin.sol

```diff
@@ -438,41 +442,92 @@ contract VaultAdmin is IVaultAdmin, VaultCommon, Authentication {
         issuedShares = wrappedToken.convertToAssets(amountWrappedRaw) + amountUnderlyingRaw;
 
-        if (_bufferAssets[wrappedToken] == address(0)) {
-            // Buffer is not initialized yet, so we initialize it.
+        _ensureMinimumTotalSupply(issuedShares);
 
-            // Register asset of wrapper, so it cannot change.
-            _bufferAssets[wrappedToken] = underlyingToken;
+        issuedShares -= _MINIMUM_TOTAL_SUPPLY;
+        _mintBufferShares(wrappedToken, sharesOwner, issuedShares);
+        _mintMinimumBufferSupplyReserve(wrappedToken);
 
-            // Burn MINIMUM_TOTAL_SUPPLY shares, so the buffer can never go back to zero liquidity
-            // (avoids rounding issues with low liquidity).
-            _bufferTotalShares[wrappedToken] = _MINIMUM_TOTAL_SUPPLY;
-            issuedShares -= _MINIMUM_TOTAL_SUPPLY;
-        } else if (_bufferAssets[wrappedToken] != underlyingToken) {
-            // Asset was changed since the first bufferAddLiquidity call.
-            revert WrongWrappedTokenAsset(address(wrappedToken));
-        }
+        emit LiquidityAddedToBuffer(wrappedToken, amountUnderlyingRaw, amountWrappedRaw);
+    }
 
-        bytes32 bufferBalances = _bufferTokenBalances[wrappedToken];
+    /// @inheritdoc IVaultAdmin
+    function addLiquidityToBuffer(
+        IERC4626 wrappedToken,
+        uint256 amountUnderlyingRaw,
+        uint256 amountWrappedRaw,
+        address sharesOwner
+    )
+        public
+        onlyVaultDelegateCall
+        onlyWhenUnlocked
+        whenVaultBuffersAreNotPaused
+        withInitializedBuffer(wrappedToken)
+        nonReentrant
+        returns (uint256 issuedShares)
+    {
+        // Check wrapped token asset correctness.
+        address underlyingToken = wrappedToken.asset();
+        _ensureCorrectBufferAsset(wrappedToken, underlyingToken);
 
-        // Adds the issued shares to the total shares of the liquidity pool.
-        _bufferLpShares[wrappedToken][sharesOwner] += issuedShares;
-        _bufferTotalShares[wrappedToken] += issuedShares;
+        // Take debt for assets going into the buffer (wrapped and underlying).
+        _takeDebt(IERC20(underlyingToken), amountUnderlyingRaw);
+        _takeDebt(wrappedToken, amountWrappedRaw);
 
+        // Update buffer balances
+        bytes32 bufferBalances = _bufferTokenBalances[wrappedToken];
         bufferBalances = PackedTokenBalance.toPackedBalance(
             bufferBalances.getBalanceRaw() + amountUnderlyingRaw,
             bufferBalances.getBalanceDerived() + amountWrappedRaw
         );
-
         _bufferTokenBalances[wrappedToken] = bufferBalances;
 
-        _takeDebt(IERC20(underlyingToken), amountUnderlyingRaw);
-        _takeDebt(wrappedToken, amountWrappedRaw);
+        // Amount of shares to issue is the total underlying token that the user is depositing.
+        issuedShares = wrappedToken.convertToAssets(amountWrappedRaw) + amountUnderlyingRaw;
```

Comment:
This line probably requires a function since it now repeats for buffer initialization and buffer addLiquidity.

Reply:
That would help 'reusability', but it introduces one more hop that sounds a bit unnecessary given that it's a single line. If it was more than one line I'd be more inclined to put it into a function, but if it's just for this I think it's more readable like this.